Projektskabelon-master/src/main/java/FunctionLayer/OverlayMaterialCalculator.java

```diff
@@ -54,20 +54,22 @@ public static Material screwSparOneWall(Wall wall) {
             fyr.setComment("Vertical framing");
             fyr.setSize(height);
             fyr.setAmount(1);
+            fyr.setPrice(LogicFacade.getPrice(fyr.getId()));
             fyrs.add(fyr);
         }
         return fyrs;
     }
 
     //..................SCREWS FOR FYR......................//
-    public static Material screwFYROneWall(Wall wall) {
+    public static Material screwFYROneWall(Wall wall) throws LoginSampleException {
         Material screwFyr = new Material();
         screwFyr.setName("BASIC SKRUE 5,0X40MM");
         int fyrQuantity = OverlaySizeCalculator.fyrQuantityOnWall(wall);
         int spaerQuantity = OverlaySizeCalculator.spaerOnOneWall(wall);
         screwFyr.setSize(OverlaySizeCalculator.screwFyr(fyrQuantity, spaerQuantity));
         screwFyr.setComment("Til montering af vertical framing");
         screwFyr.setAmount(1);
+        screwFyr.setPrice(LogicFacade.getPrice(screwFyr.getId()));
```

Comment:
If you set the price here, it will be calculated for the needed size, not for the available one. I calculate de available sizes first in function layer